Project team for the Rust Vision Doc 2025 effort

nikomatsakis opened the pull request:
[Context](https://rust-lang.zulipchat.com/#narrow/channel/486265-vision-doc-2025/topic/vision.20doc.20plan/near/505781528); [this hackmd contains charter, expectations, and other info](https://hackmd.io/@rust-vision-doc/SyfqOGGhye).

cc @jackh726 

### Charter

To author the Rust 2025 Vision RFC. See the [Project Goal](https://rust-lang.github.io/rust-project-goals/2025h1/rust-vision-doc.html) for more details.

### Membership plan

Initial leads, charged with generally running the group and determining members and membership expectations.

* nikomatsakis
* jackh726

### Expectations of membership

The expectation of a team member is that you will be an active participant in gathering and analyzing data. This means that you'll setup and drive at least 2 or 3 interviews and discussions during April -- can of course be way more! There are no pre-defined focus areas, it's up to you to identify the kinds of things you want to focus on.

Time expectation: 2-3h/wk

* Identify at least 2 or 3 interviews you would like to lead
    * Attend others as shadow interviewer
* Conduct interviews in alignment with our guidelines
* Attend regular sync meetings to discuss
* Actively paticipate in editoral discussions

ehuss commented on teams/project-vision-doc-2025.toml:
```diff
@@ -0,0 +1,25 @@
+name = "project-vision-doc-2025"
+kind = "project-group"
+subteam-of = "leadership-council"
```
I think this should probably be:

```suggestion
subteam-of = "launching-pad"
```

We don't want any teams (other than the top-level teams) directly linked under the council. I think as we move towards having the launching pad (name TBD) be a home for cross-cutting teams, this somewhat fits. Or, we find some different structure in the future, in which case we can move things.

ehuss commented on people/ernestkissiedu.toml:
```diff
@@ -0,0 +1,4 @@
+name = "Ernest Kissiedu "
```
```suggestion
name = "Ernest Kissiedu"
```

nikomatsakis commented:
@ZuseZ4 no-- I don't understand that, I thought all those docs were public.

nikomatsakis commented:
Heads up that we updated the charter and expectations of membership and I plan to add a few more folks to the list.

nikomatsakis commented:
@ZuseZ4 I've inlined the charter into the PR description, seems better anyhow.

Mark-Simulacrum commented:
@nikomatsakis Are we expecting a T-council decision to gate merging this? (I don't have a _super_ strong opinion, but it feels a bit odd to allow almost top-level team creation without _some_ team approving it. In this case it's mostly longstanding project members so I'm not too worried, but still :)

I'll mark this PR as draft, but please poke if we think this is only waiting on team repo admins...

jackh726 commented:
I hadn't quite thought about if this is needs some extra approval or not. I was just waiting on Niko to update the member list here and add a repo before merging.

nikomatsakis commented:
@Mark-Simulacrum I believe the general approval was given here when the project goal was approved, so I don't feel the need for a formal check-off...

https://rust-lang.github.io/rust-project-goals/2025h1/rust-vision-doc.html#ownership-and-team-asks

nikomatsakis commented:
@jackh726 I thought I *had* pushed that...huh

nikomatsakis commented:
OK, pushed!

nikomatsakis commented:
Rebased and updated. Given that [leadership council approved this concept](https://github.com/rust-lang/leadership-council/issues/165#issuecomment-2761914449), can we get this merged? Would be helpful.

Mark-Simulacrum commented:
But looks like CI is failing... happy to get pinged on Zulip once that's fixed.
